client/client-lib/src/lib.rs

```diff
@@ -871,32 +854,33 @@ impl Client<GatewayClientConfig> {
     /// Note though that extended periods of staying offline will result in loss of funds anyway if
     /// the client can not claim the respective contract in time.
     pub fn save_outgoing_payment(&self, contract: OutgoingContractAccount) {
-        self.context
-            .db
-            .insert_entry(
-                &OutgoingContractAccountKey(contract.contract.contract_id()),
-                &contract,
-            )
-            .expect("DB error");
+        let mut dbtx = self.context.db.begin_transaction();
+        dbtx.insert_entry(
+            &OutgoingContractAccountKey(contract.contract.contract_id()),
+            &contract,
+        )
+        .expect("DB error");
+        dbtx.commit_tx().expect("DB Error");
```

Comment:
The client might do things in parallel, is it actually true that there can be no DB errors? `expect("DB error")` was used in places where it essentially means "our DB isn't reachable", which for an embedded DB essentially means your hard drive went missing/is on fire imo. If an error can actually reasonably occur we should handle it instead.

Reply:
Yes, you're right, I think it is possible for conflicts to happen here. I'll revisit these and try to handle them properly.